pkg/k8s/watchers/subscriber/ciliumnode.go

```diff
@@ -49,46 +49,31 @@ func (l *CiliumNodeChain) Register(s CiliumNode) {
 func (l *CiliumNodeChain) OnAddCiliumNode(node *ciliumv2.CiliumNode) error {
 	l.RLock()
 	defer l.RUnlock()
-	errs := []error{}
+	var err error
 	for _, s := range l.subs {
-		if err := s.OnAddCiliumNode(node); err != nil {
-			errs = append(errs, err)
-		}
+		err = errors.Join(s.OnAddCiliumNode(node))
```

Comment:
Doesn't this only return the error from last subscriber since we overwrite `err`?
`err = errors.Join(err, s.OnAddCiliumNode(node))` would probably not work since we get nested `joinError`s. Perhaps just do the same as the original except use `errors.Join(errs...)` instead of `fmt.Errorf`.